src/date.ts

```diff
@@ -204,4 +187,12 @@ export class _Date {
 
     return this.faker.random.arrayElement(source);
   }
+
+  private createDate(date: string | Date) {
+    if (typeof date !== 'undefined') {
```

Comment:
For sake of simplicity, I would suggest following:

```suggestion
    if (date != null) {
```